### ISSUE TITLE
Add Webform Views Integration module to GovCMS

### ISSUE THREAD

**Issue by StephenOllman** [govCMS/GovCMS#1315]
** What value does this module/package add to GovCMS?
**Allows site administrators to create custom reports of webform submissions. Currently a site administrtaor has to export submission data to a CSV file (or similar) in order to creae a report.**

** Is the module Drupal 10 compatible?
**8.x-5.5 - Drupal ^10 ||  ^11**

** Please provide a brief outline of what this module does.
**Use Webform Views Integration module to expose Webform submissions to Views (Webform is a module for making forms and surveys in Drupal). This wiull allow administrator the ability to create custom views to generate filterable reports based on webform submitted data.**

** Who does this module benefit:
[ ] end users
[ ] content editors
**[X] site builders**
[ ] themers
[ ] developers
**[X] site senior management**
 
** How could you provide/replicate the functionality of this module using alternative methods, eg in your theme?
**Export webform submission data to a CSV and use something like Excel to create custom reports.**

** If this module styles or alters HTML or JavaScript output, can the functionality be provided via the theme? What alternatives have you considered.
**N/A**

** What is the maintenance and support status of the module. Describe the issue queue activity.
**Last updated 23 November 2024. Active issues register.**

** What permissions are needed to utilise the module (and are any new permissions provided by the module)?
**Unknown but probably relies on permissions for both Webform and Views.**

** Does the module modify the database structure and/or store additional metadata on nodes or other entities? If so, why? What are the risks for future updates?
**Not that I am aware of. It primarily exposes webform database data to the Views module.**

** Is the module designed to capture anonymous user data?
**No**

** Is the output of the module typically fully cacheable? Would the inclusion of this module potentially render pages uncacheable.
**Can be cached by the Views module functionality.**

** What is your assessment of the quality of this module, the contribution history of the module's maintainers, and the uptake of the module within the Drupal community?
**Module has been around since 2009 and has an active community.**

** Additional information

**Comment by thisisalistairsaccount** [comment]
The module can be found at: https://www.drupal.org/project/webform_views

If you use Views on a website, you will know that you can customise and order things for both end users but also for administration spaces. Think like the content or media lists, they are views. However, something that Views doesn't do out of the box, is work with webform submissions. This module allows for that.

The question to the community is, do you think this is valuable to be included?

If you are doing webforms, as @StephenOllman has mentioned, you can see submissions in a very simple format within the website. You can also take downloads of those webform submissions and manage them within Excel, LibreOffice Calc and other software that can handle structured data.

**Comment by kobus-vanwyk** [comment]
Hi community, does anyone have any feedback or views on this module (webform_views) or other considerations? Seeking views from the GovCMS community.

**Comment by StephenOllman** [comment]
I acknowledge this might be an edge case requirement and as such if no one else see's a benefit, I'm happy to remove this module nomination.

**Comment by kobus-vanwyk** [comment]
No need to remove the module nomination.
The question to the community is, do you think [webform_views](https://www.drupal.org/project/webform_views) is valuable to be included?

**Comment by pstampy** [comment]
Hey @StephenOllman I am going to close this request as there hasn't been much community input. Happy to discuss :)

**Comment by StephenOllman** [comment]
Agreed. It's been over 8 months since posting with no community engagement. Edge case functionality.
